research/src/pages/popup/popup.proposal.mdx

```diff
@@ -0,0 +1,434 @@
+3. Otherwise, if the _invoker_ is not null, starting with the _invoker_, walk the
+   [flat tree](https://drafts.csswg.org/css-scoping/#flattening) to determine whether the _subject_
+   is an ancestor to the _invoker_. If this condition is met, then return.
+4. Otherwise, if the _anchor element_ is not null, starting with the _anchor element_, walk the
+   [flat tree](https://drafts.csswg.org/css-scoping/#flattening) to determine whether the _subject_
+   is an ancestor to the _anchor element_. If this condition is met, then return.
+5. If _start node_ is not null, starting with the _start node_, walk the
+   [flat tree](https://drafts.csswg.org/css-scoping/#flattening) to determine whether the _subject_
+   is the _start node_ or an ancestor to the _start node_. If either condition is met, then return.
+6. Otherwise, for this _subject_, run the
+   [hiding a `popup` element steps](#hiding-popup-steps), with _subject_ as _candidate subject_.
+
+<h3 id="hiding-popup-steps">
+  Hiding a <code>popup</code> element steps
+</h3>
+
+The **hiding a `popup` element steps**, given a `popup` element _candidate subject_ and an optional
+element _invoker_, are as follows:
+
+1. Remove the _candidate subject_ from the
+   [node document](https://dom.spec.whatwg.org/#concept-node-document)'s [popup stack](#popup-stack).
+2. Set the `open` IDL attribute on the _candidate subject_ to `false`.
+3. Set the _candidate subject_'s _open state_ to `false`.
+4. If applicable, empty the _candidate subject_'s _invoker_.
+5. [Queue an element task](https://html.spec.whatwg.org/multipage/webappapis.html#queue-an-element-task)
+   on the [user interaction task source](https://html.spec.whatwg.org/multipage/webappapis.html#user-interaction-task-source)
+   given the _candidate subject_ element to
+   [fire an event](https://dom.spec.whatwg.org/#concept-event-fire)
+   named `hide` at _candidate subject_.
+
+---
+
+## HTML Incubation Text: User Interaction > Focus > Processing Model
+
+<p class="note">
+  The following text to be inserted into the
+  <a href="https://html.spec.whatwg.org/multipage/interaction.html#focus-processing-model)">
+    get the focusable area steps
+  </a>.
+</p>
+
+**If _focus target_ is a `popup` element**
+
+1. If the `delegatesfocus` attribute is specified on _focus target_, return the first focusable
+   descendent element of _focus target_, in
+   [tree order](https://dom.spec.whatwg.org/#concept-tree-order), that is not inert.
+2. If there is no such descendent, but `delegatesfocus` is specified, return the _focus target_.
+3. Otherwise, if the `autofocus` attribute is specified on _focus target_, return the _focus target_.
+4. Otherwise, if the `autofocus` attribute is specified on a descendent element of _focus target_,
+   return the first such descendent element of _focus target_, in
+   [tree order](https://dom.spec.whatwg.org/#concept-tree-order), that is not inert.
```

Comment:
Interesting. This allows autofocus to focus elements which are not normally focusable (as long as they are not inert). Is that intended behavior?

The normal autofocus behavior does not do this. It performs "getting the focusable area" on the element where autofocus is specified, which might return the element itself, or might return null. If it returns null, then the attribute does nothing.